cumulusci/tasks/metadata/modify.py

```diff
@@ -1,52 +1,113 @@
+            ),
             "required": True,
         },
         "chdir": {
             "description": "Change the current directory before running the replace"
         },
+        "output_style": {
+            "description": "Output style to use: 'salesforce' or 'simple'",
+            "required": False,
+        },
     }
 
+    def _init_options(self, kwargs):
+        super()._init_options(kwargs)
+        self.chdir = self.options.get("chdir")
+        self.elements = self.options["elements"]
+        self.output_styles = self.options.get("output_style", "")
+
     def _run_task(self):
-        chdir = self.options.get("chdir")
-        if chdir:
-            self.logger.info("Changing directory to {}".format(chdir))
-        with cd(chdir):
-            for element in self.options["elements"]:
-                self._process_element(element)
-
-    def _process_element(self, element):
+        if self.chdir:
+            self.logger.info("Changing directory to {}".format(self.chdir))
+        with cd(self.chdir):
+            for element in self.elements:
+                self._process_element(element, self.output_styles)
+
+    def _process_element(self, step, output_style):
         self.logger.info(
-            "Removing elements matching {xpath} from {path}".format(**element)
+            "Removing elements matching {xpath} from {path}".format(**step)
         )
-        for f in glob.glob(element["path"]):
+        for f in glob.glob(step["path"], recursive=True):
+            self.logger.info(f"Checking {f}")
             with open(f, "rb") as fp:
                 orig = fp.read()
             root = ET.parse(f)
-            res = root.findall(
-                element["xpath"].replace(
-                    "ns:", "{http://soap.sforce.com/2006/04/metadata}"
-                )
+            res = root.xpath(
+                step["xpath"],
+                namespaces={
+                    "ns": "http://soap.sforce.com/2006/04/metadata",
+                    "re": "http://exslt.org/regular-expressions",
+                },
             )
+            self.logger.info(f"Found {len(res)} matching elements")
             for element in res:
                 element.getparent().remove(element)
-            processed = (
-                b'<?xml version="1.0" encoding="UTF-8"?>\n'
-                + ET.tostring(root, encoding="utf-8")
-                + b"\n"
-            )
+
+            if output_style.lower() == "salesforce":
+                processed = bytes(salesforce_encoding(root), "utf-8")
+            else:
+                processed = (
+                    bytes(xml_encoding, encoding="utf-8")
+                    + ET.tostring(root, encoding="utf-8")
+                    + b"\n"
+                )
+
             if orig != processed:
                 self.logger.info("Modified {}".format(f))
                 with open(f, "wb") as fp:
                     fp.write(processed)
+
+
+def salesforce_encoding(xdoc):
```

Comment:
I didn't change it to be the default or required because it would technically be a backwards-incompatible change. Technically.

We have a few options:
 * always SF encode
 * make it the default and people who notice breaking will just change their code back with the option
 * start some transition process involving deprecating configs that don't pass it
 * do what I did already and call it a day

Reply:
@prescod I'm not concerned about backwards compatibility in this case. This task isn't even configured in the default cumulusci.yml, and I don't find any uses of it in the SalesforceFoundation github organization. And it's only a change in formatting, not semantics.

Reply:
Okay so I started refactoring, but I'll just warn you that in some places we use 

import xml.etree.ElementTree as ET

and in others:

import lxml.etree as ET

So when I unify them (as lxml.etree), there may be some unintended consequences because it is a "compatible" but different implementation. I'm already having to fix some test cases...

Reply:
@prescod oh, right. I care more about showing error messages that identify what file was being processed than I care about consistency in which xml library we use (or at least I'm more willing to defer solving the latter)

Reply:
Okay, output_style is removed now.

Reply:
I still think a docstring would be helpful for our future selves.